drivers/usb/device/usb_dc_nrfx.c

```diff
@@ -1676,7 +1684,6 @@ int usb_dc_ep_write(const u8_t ep, const u8_t *const data,
 		return 0;
 	}
 
-	int result = 0;
```

Comment:
Is results needed only for one check? I would use 2 returns instead.

```
if (err != NRFX_SUCCESS) {
		ep_ctx->write_in_progress = false;
		LOG_ERR("nRF USBD write error: %d", (u32_t)err);
                return -EIO;
	}
	k_mutex_unlock(&ctx->drv_lock);
return 0;
```

Reply:
I did not look at it closely, just fixed coding style issues and moved it to the beginning of the block.

Reply:
@finikorg
> Is results needed only for one check? I would use 2 returns instead.

Then the mutex would not be unlocked in the `-EIO` case.

Reply:
Ah, sorry, need to put mutex_unlock before return.